src/ChainSafe.Gaming.WalletConnect/WalletConnectCustomProvider.cs

```diff
@@ -29,18 +30,24 @@ public class WalletConnectCustomProvider : IWalletConnectCustomProvider, ILifecy
         private readonly IOperatingSystemMediator operatingSystem;
         private readonly ILogWriter logWriter;
         private readonly WalletConnectConfig config;
+        private readonly IAnalyticsClient analyticsClient;
+        private readonly IChainConfig chainConfig;
+        private readonly IProjectConfig projectConfig;
 
         /// <summary>
         /// Initializes a new instance of the <see cref="WalletConnectCustomProvider"/> class.
         /// </summary>
         /// <param name="config">Wallet connect configuration used to pass values to this provider.</param>
         /// <param name="operatingSystem">Operating system mediator used for passing platform information and opening a deeplink.</param>
         /// <param name="logWriter">Log writer used for logging messages to platform.</param>
-        public WalletConnectCustomProvider(WalletConnectConfig config, IOperatingSystemMediator operatingSystem, ILogWriter logWriter)
+        public WalletConnectCustomProvider(WalletConnectConfig config, IOperatingSystemMediator operatingSystem, ILogWriter logWriter, IAnalyticsClient analyticsClient, IChainConfig chainConfig, IProjectConfig projectConfig)
```

Comment:
I was thinking we can add `IChainConfig chainConfig, IProjectConfig projectConfig` as dependencies to the analytics client itself, so that we don't have to add them to each CaptureEvent call. WDYT?

Reply:
Makes sense will do that right away

Reply:
Did that TY!